### ISSUE TITLE
Following problem

### ISSUE THREAD

**Issue by socialhammer** [mgp25/Instagram-API#282]
Hello.

In some cases, sending a follow request, it doesn't happen, it is seen in the flag 
friendship_status -> following, there has to be 1

**Example:**

```
POST /api/v1/friendships/create/3247358133/ HTTP/1.1
Host: i.instagram.com
User-Agent: Instagram 8.4.0 Android (18/4.3; 320dpi; 720x1280; Xiaomi; HM 1SW; armani; qcom; en_US)
POST:
ig_sig_key_version=4&signed_body=ebf3210730c85387ab92a0cf1091448c8e6b6c172db479de014e4adf23a2da35.%7B%22_uuid%22%3A%223166a491-1373-4177-9f12-b4c5431e15a4%22%2C%22_uid%22%3A2291259951%2C%22user_id%22%3A3247358133%2C%22_csrftoken%22%3A%228Jz32foo8R1UAhkQQs8GwGkj4rwsnG6l%22%7D
```

**Answer:** 
```
$Following = Array
(
[status] => ok
[friendship_status] => Array
(
[incoming_request] => 
[followed_by] => 
[outgoing_request] => 
[following] => 
[blocking] => 
[is_private] => 
)
)
```

In result we can't do following. What is problem?

Sometimes for this account following working normal but sometimes we get this problem. Pauses between requests didn't help.

**Comment by MMrR0b0TT** [comment]
Check if the user is private, in this case, he'll need to accept your request to get following true.

**Comment by socialhammer** [comment]
No, this is normal accounts:
https://www.instagram.com/kitriks/
https://www.instagram.com/liz__kaa98/

**Comment by huttarichard** [comment]
@socialhammer 
I believe this is sort of anti-spam security (more user friendly). 
If you do this through official app you will get same result (iphone & android), make follow and refresh profile and you are not following. 
You can try as workaround to change proxy, works for me ;).

**Comment by huttarichard** [comment]
You can actually detect this antispam control by double check on friendship status

**Comment by socialhammer** [comment]
huttarichard, what do you mean about "double check on friendship status" ?

**Comment by huttarichard** [comment]
@socialhammer I did, I have my hands on this issue for a while. You can try to do follow and after sec do an friendship status request 
https://github.com/mgp25/Instagram-API/blob/master/src/Instagram.php#L1572

you will get result about friendship status and if you are actually really following or not.

**Comment by socialhammer** [comment]
So may be this is anti-spam security. Without proxy changing what can I do?

**Comment by huttarichard** [comment]
@socialhammer does not matter what it is, what matters is result. If proxy helps then you have a solution for this problem. I would say wait until you can do follow again or change an address or change account otherwise there is really nothing you can do about this :).

**Comment by mgp25** [comment]
Yes, it seems to be a protection to avoid bulk request. BTW, dont bulk request, its against the EULA

Regards

**Comment by lucky1412** [comment]
ig_sig_key_version=4&signed_body=2363c77e03f96f27897fc1ff180c7850a9648927fde484b4dfc1fc6836cd9abd.{"_csrftoken":"aJRNCF9gHQTTa1BZWoijyuQJtpjhSib4","_uid":"1659247553","_uuid":"841da2b8-556a-4888-9a7d-9b1eb5a13605","user_id":"266928623"}

but is gives me {"status": "fail", "message": "Please update your Instagram app to continue following."}
please help me out 

thanks 
Santosh

**Comment by huttarichard** [comment]
@lucky1412 you have wrong key ;)

**Comment by lucky1412** [comment]
thanks huttarichard i solved it there is big mistake in my key .. so thanks for ur acknowledgment  :)

**Comment by AndiHamolli** [comment]
I am having the same issue 
@socialhammer @huttarichard @lucky1412 @mgp25
I noticed that issue as well happening with my account today. Don't know what is causing it. Perhaps I am following too often, but when I set $debug true i don't get any error code.

and if the user is private it should print:
`[outgoing_request] => ` should be 1.

but it doesn't do anything, no error, no msg that I am doing follows too often (15 follows/unfollos for 30 min) or anything. It just prints the same empty array

[friendship_status] => InstagramAPI\Response\Model\FriendshipStatus Object
        (
            [following] => 
            [followed_by] => 
            [incoming_request] => 
            [outgoing_request] => 
            [is_private] => 
            [is_blocking_reel] => 
            [is_muting_reel] => 
            [blocking] => 
            [is_bestie] => 
        )
